hawkular-wildfly-monitor/src/main/java/org/hawkular/agent/monitor/cmd/AbstractRemoveModelNodeCommand.java

```diff
@@ -0,0 +1,144 @@
+
+/**
+ * @author <a href="https://github.com/ppalaga">Peter Palaga</a>
+ */
+public abstract class AbstractRemoveModelNodeCommand<REQ extends ResourcePathRequest, RESP extends ResourcePathResponse>
+        implements Command<REQ, RESP> {
+    private static final MsgLogger log = AgentLoggers.getLogger(AbstractRemoveModelNodeCommand.class);
+
+    private final String entityType;
+
+    public AbstractRemoveModelNodeCommand(String entityType) {
+        super();
+        this.entityType = entityType;
+    }
+
+    @Override
+    public BasicMessageWithExtraData<RESP> execute(REQ request, BinaryData ignored, CommandContext context)
+            throws Exception {
+
+        String rawResourcePath = request.getResourcePath();
+        log.infof("Received request to remove [%s] given by inventory path [%s]", entityType, rawResourcePath);
+
+        MonitorServiceConfiguration config = context.getMonitorServiceConfiguration();
+
+        // Based on the resource ID we need to know which inventory manager is handling it.
+        // From the inventory manager, we can get the actual resource.
+        CanonicalPath canonicalPath = CanonicalPath.fromString(rawResourcePath);
+
+        String resourceId = canonicalPath.ids().getResourcePath().getSegment().getElementId();
+        ResourceIdParts idParts = InventoryIdUtil.parseResourceId(resourceId);
+        String modelNodePath = idParts.getIdPart();
+
+        RESP response = createResponse();
+        MessageUtils.prepareResourcePathResponse(request, response);
+
+        DatasourceJBossASClient dsc = null;
+        try {
+            String managedServerName = idParts.getManagedServerName();
+            ManagedServer managedServer = config.managedServersMap.get(new Name(managedServerName));
+            validate(request, managedServerName, managedServer);
+
+            validate(modelNodePath, request);
+            DMRInventoryManager inventoryManager = context.getDiscoveryService().getDmrServerInventories()
+                    .get(managedServer);
+            dsc = new DatasourceJBossASClient(inventoryManager.getModelControllerClientFactory().createClient());
+
+            Address addr = Address.parse(modelNodePath);
+            dsc.remove(addr);
+
+            afterModelNodeRemoved(request, response);
+
+        } catch (Exception e) {
+            response.setStatus(ResponseStatus.ERROR);
+            String msg = String.format("Could not remove [%s] [%s]: %s", entityType, rawResourcePath, e.getMessage());
+            response.setMessage(msg);
+        } finally {
+            if (dsc != null) {
+                try {
+                    dsc.close();
+                } catch (Exception e) {
+                    log.errorf(e, "Could not close a DatasourceJBossASClient");
+                }
+            }
+        }
+
+        return new BasicMessageWithExtraData<>(response, null);
+
+    }
+
+    /**
+     * @param modelNodePath
+     * @param request
+     */
```

Comment:
fill in javadoc to explain what this is validating  - same with the empty javadocs in the method below